### DISCUSSION TITLE
Capacitor 3: Run livereload with with ssl

### DISCUSSION THREAD

**Discussion by santosbj** [ionic-team/capacitor#5136] [Q&A]
I need to run the capacitor live reload with ssl, to deal with geolocation application

ionic cap run -l --external --ssl

In capacitor 2 to make this work, I need add the code  to MainActivity.java. Otherwise, the self signed certificate is not accepted.

@Override
public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    this.init(savedInstanceState, new ArrayList<Class<? extends Plugin>>() {{
        // Additional plugins you've installed go here
        // Ex: add(TotallyAwesomePlugin.class);
    }});

    if (BuildConfig.DEBUG) {
        this.bridge.getWebView().setWebViewClient(new BridgeWebViewClient(this.bridge) {
            @Override
            public void onReceivedSslError(WebView view, final SslErrorHandler handler, SslError error) {
                handler.proceed();
            }
        });
    }
}

But, this not work on capacitor 3.

Any idea of how to pass this situation?

**Comment by Cselt** [comment]
I had the same error while I was trying to update my project to capacitor 3.
As it looks in the `onCreate` callback the `getWebView` function returns null (not yet initialized)

I tried to use `onPostCreate` callback, but for some reason that got never invoked.

So my solution works with `onResume`:

```java
public class MainActivity extends BridgeActivity {

  private boolean webViewInitialized = false;

  @Override
  public void onResume() {
    super.onResume();

    if(webViewInitialized) return;

    getBridge().getWebView().setWebViewClient(new BridgeWebViewClient(getBridge()) {
      @Override
      public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError error) {
        handler.proceed();
      }
    });
    webViewInitialized = true;
  }
}
```

I added a helper variable `webViewInitialized` because onResume callback can be invoked many times. See [Android docs](https://developer.android.com/guide/components/activities/activity-lifecycle#onresume) for details.
